dist/js/header/headerController.js

```diff
@@ -1,24 +1,100 @@
 export default class HeaderController {
-    constructor(headerMenuContainer, headerHeading) {
-        this.headerMenuContainer = headerMenuContainer;
-        this.headerHeading = headerHeading;
+  constructor(headerMenuContainer, headerHeading) {
+    this.headerMenuContainer = headerMenuContainer;
+    this.headerHeading = headerHeading;
+    this.previousYCoordinate = 0;
+    this.scrollEventExecutionFlag = false;
+    this.headerMenuContainerSize = 0;
+    this.menuShowFlag = false;
+  }
+
+  init() {
+    this.getHeaderElementSize();
+    this.pageScrollListener();
+  }
+
+  pageScrollListener() {
+    document.addEventListener('scroll', () => {
+      if (!this.scrollEventExecutionFlag) {
+        requestAnimationFrame(this.scrollEvent.bind(this));
+        this.scrollEventExecutionFlag = true;
+      }
+    });
+  }
+
+  scrollEvent() {
+    const scrollTopSafeSpace = 50; // 50px scroll top to show sticky menu
+    const docElement = document.documentElement;
+    const scrollBounceValueForSafari = docElement.scrollHeight - docElement.clientHeight - docElement.scrollTop;
+    const scrollBounce = scrollBounceValueForSafari < 0; // boolean scroll bounce value
+
+    const deltaOfPreviousAndCurrentCoords = this.previousYCoordinate - window.pageYOffset;
+    const scrollTopValueWithSafeInterval = window.pageYOffset + scrollTopSafeSpace;
+
+    // Check bounce (for Safari)
+    if (scrollBounce) {
+      this.scrollEventExecutionFlag = false; // set event execution flag to false
+      return;
+    }
+
+    // Checking conditions to fix the menu
+    if (window.pageYOffset >= this.headerMenuContainerSize) {
+      this.headerMenuShouldStick();
+    } else {
+      this.headerMenuShouldStickOut();
+    }
+
+    // Checking conditions for dynamic menu display
+    if (deltaOfPreviousAndCurrentCoords >= 0 && this.menuShowFlag) { // scroll top after menu was shown
+      this.headerMenuShouldShow();
+    } else if (this.previousYCoordinate <= scrollTopValueWithSafeInterval && pageYOffset >= this.headerMenuContainerSize) { // scroll bottom or check scroll more than default menu location
+      this.headerMenuShouldHide();
+    } else { // other
+      this.headerMenuShouldShow();
     }
```

Comment:
Looks like first `if` or last `else` is redundant

Reply:
fixed